src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,112 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidInputDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static User user;
+    private static final int VALID_AGE = 18;
+    private static final String VALID_PASSWORD = "userPassword";
+    private static final String VALID_LOGIN = "validLogin";
+    private static final String SHORT_PASSWORD = "abc";
+
+    @Test
+    void register_userIsNull_NotOk() {
+        User newUser = null;
+        assertThrows(InvalidInputDataException.class, () ->
+                registrationService.register(newUser)
+        );
+    }
+
+    @Test
+    void register_loginIsNull_NotOk() {
+        user.setLogin(null);
+        Assertions.assertThrows(InvalidInputDataException.class,
+                () -> registrationService.register(user), "Login can't be null"
+        );
+    }
+
+    @Test
+    void register_userWithSuchLoginAlreadyExist_NotOk() {
+        registrationService.register(user);
```

Comment:
Add user to the db with low-level methods. You can not set up you db in a test with the same methods you test